packages/botfuel-dialog/tests/messages/bot-table-message.test.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2017 - present, Botfuel (https://www.botfuel.io).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const BotTableMessage = require('../../src/messages/bot-table-message');
+const MessageError = require('../../src/errors/message-error');
+
+describe('BotTableMessage', () => {
+  test('should throw an exception when malformed', async () => {
+    expect(() => new BotTableMessage(null)).toThrow(MessageError);
+
+    // no rows
+    expect(() => new BotTableMessage({ schema: [] })).toThrow(MessageError);
+
+    // no schema
+    expect(() => new BotTableMessage({ rows: [] })).toThrow(MessageError);
+  });
```

Comment:
You could add tests for:

- A good table message
- A message that has both `schema` and `rows` but they are not arrays (so that the lines
```javascript
    this.validateArray('data.schema', this.data.schema);
    this.validateArray('data.rows', this.data.rows);
```
 are covered by tests)